Fails to compile with GHC 9.2.1

> [13 of 14] Compiling DBus.Generation  ( lib/DBus/Generation.hs, dist/build/DBus/Generation.o, dist/build/DBus/Generation.dyn_o )
> 
> lib/DBus/Generation.hs:151:24: error:
>     • Couldn't match expected type ‘Pat’
>                   with actual type ‘[Pat] -> Pat’
>     • Probable cause: ‘ConP’ is applied to too few arguments
>       In the expression: ConP 'Just [VarP name]
>       In an equation for ‘makeJustPattern’:
>           makeJustPattern name = ConP 'Just [VarP name]
>     |
> 151 | makeJustPattern name = ConP 'Just [VarP name]
>     |                        ^^^^^^^^^^^^^^^^^^^^^^
> 
> lib/DBus/Generation.hs:151:36: error:
>     • Couldn't match expected type ‘Type’ with actual type ‘Pat’
>     • In the expression: VarP name
>       In the second argument of ‘ConP’, namely ‘[VarP name]’
>       In the expression: ConP 'Just [VarP name]
>     |
> 151 | makeJustPattern name = ConP 'Just [VarP name]
>     |                                    ^^^^^^^^^
> cabal: Failed to build dbus-1.2.21

--- a/.github/workflows/cabal.yaml
+++ b/.github/workflows/cabal.yaml
@@ -6,7 +6,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        ghc: ['8.8', '8.10', '9.0']
+        ghc: ['9.2']
     steps:
       - uses: actions/checkout@v2
       - uses: haskell/actions/setup@v1

--- a/dbus.cabal
+++ b/dbus.cabal
@@ -1,5 +1,5 @@
 name: dbus
-version: 1.2.22
+version: 1.2.23
 license: Apache-2.0
 license-file: license.txt
 author: John Millikin <john@john-millikin.com>
@@ -94,7 +94,7 @@ library
     , parsec
     , random < 1.3
     , split
-    , template-haskell < 2.18.0.0
+    , template-haskell >= 2.18 && < 2.19
     , text
     , th-lift < 0.9
     , transformers

--- a/lib/DBus/Generation.hs
+++ b/lib/DBus/Generation.hs
@@ -148,7 +148,7 @@ makeFromVariantApp :: Name -> Exp
 makeFromVariantApp name = AppE (VarE 'T.fromVariant) $ VarE name
 
 makeJustPattern :: Name -> Pat
-makeJustPattern name = ConP 'Just [VarP name]
+makeJustPattern name = ConP 'Just [] [VarP name]
 
 mapOrHead ::
   (Num a, Eq a) => a -> (t -> b) -> [t] -> ([b] -> b) -> b

